File: ALGORITMOS E PROGRAMAÇÃO/C/Lista 2/questao17.c
#include <stdio.h>
#include <locale.h>

int main();

void apresentacao()
{
	printf("Questão 17:\nVocê tem um desejo incontrolável de descobrir quantos dias você já viveu em toda a sua vida,\n"
		"então faça um algoritmo que leia o ano de nascimento de uma pessoa e o ano atual, depois\n"
		"calcule e escreva o nome e o número de dias, aproximados, já vividos por essa pessoa\n\n");
}

void validavalor(int status, int* valor, int i)
{
	int temp, input;
	while (status != 1 || *valor < 0)
	{
		while ((temp = getchar()) != EOF && temp != '\n');
		printf("Input inválido....\n");

		if (i == 0)
		{
			printf("Qual o seu ano de nascimento? ");
			status = scanf("%d", valor);
		}
		else if (i == 1)
		{
			printf("Qual o ano atual? ");
			status = scanf("%d", valor);
		}
	}
}

void pegaano(int* ano_nascimento, int* ano_atual)
{
	printf("Qual o seu ano de nascimento? ");
	int status = scanf("%d", ano_nascimento);
	validavalor(status, ano_nascimento, 0);

	printf("Qual o ano atual? ");
	status = scanf("%d", ano_atual);
	validavalor(status, ano_atual, 1);

	if (*ano_atual <= *ano_nascimento)
	{
		printf("Data de nacimento não pode ser maior que o ano atual;\n");
		pegaano(ano_nascimento, ano_atual);
	}
}

int calculadias(int* dias, int* ano_nascimento, int* ano_atual)
{
	*dias = (*ano_atual - *ano_nascimento) * 360;
}

void imprimeresultado(int dias)
{
	printf("Dias vividos: %d\n", dias);
}

int main()
{
	setlocale(LC_ALL, "");

	int ano_nascimento, ano_atual, dias_vividos;

	apresentacao();
	pegaano(&ano_nascimento, &ano_atual);
	calculadias(&dias_vividos, &ano_nascimento, &ano_atual);
	imprimeresultado(dias_vividos);
}

File: algoritmos_e_programacao/PYTHON/LISTA02/questao17.py


class Questao17:
    def __init__(self, ano_atual, ano_nascimento):

        self.__ehvalido = ano_atual > 0 and ano_nascimento > 0 and ano_atual > ano_nascimento
        if self.__ehvalido:
            self.__ano_atual = ano_atual
            self.__ano_nascimento = ano_nascimento
        else:
            raise ValueError("Ano atual não pode ser negativo ou menor que ano nacimento")


    def calcula_dias(self):

        self.__dias = (self.__ano_atual - self.__ano_nascimento) * 360

    def __str__(self):
        self.calcula_dias()
        return f'A quantidade de dias vividos é de {self.__dias} dias'

if __name__ == '__main__':

    ano_atual = int(input("Digite o ano atual: "))
    ano_nascimento = int(input("Digite seu ano de nascimento: "))

    print(Questao17(ano_atual, ano_nascimento))